sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -316,4 +319,68 @@ public void setParameters(KV<Integer, String> element, PreparedStatement stateme
     pipeline.run();
   }
 
+  private static class ModuloTen
+        extends DoFn<KV<Integer, String>, KV<Integer, KV<Integer, String>>> {
+    @ProcessElement
+    public void processElement(ProcessContext context) throws Exception {
+      context.output(KV.of(context.element().getKey() % 10, context.element()));
+    }
+  }
+
+  private static class GetValues
+        extends DoFn<KV<Integer, Iterable<KV<Integer, String>>>, Iterable<KV<Integer, String>>>
+        implements Serializable {
+    @ProcessElement
+    public void processElement(ProcessContext context) throws Exception {
+      context.output(context.element().getValue());
+    }
+  }
+
+  @Test
+  public void testWriteIterator() throws Exception {
+    final long rowsToAdd = 1000L;
+
+    String tableName = DatabaseTestHelper.getTestTableName("UT_WRITE_ITERATOR");
+    DatabaseTestHelper.createTable(dataSource, tableName);
+    try {
+      ArrayList<KV<Integer, String>> data = new ArrayList<>();
+      for (int i = 0; i < rowsToAdd; i++) {
+        KV<Integer, String> kv = KV.of(i, "Test");
+        data.add(kv);
+      }
+      pipeline.apply(Create.of(data))
+          .apply(ParDo.of(new ModuloTen()))
+          .apply(GroupByKey.<Integer, KV<Integer, String>> create())
+          .apply(ParDo.of(new GetValues()))
+          .apply(JdbcIO.<KV<Integer, String>>writeIterable()
+              .withDataSourceConfiguration(JdbcIO.DataSourceConfiguration.create(
+                  "org.apache.derby.jdbc.ClientDriver",
+                  "jdbc:derby://localhost:" + port + "/target/beam"))
+              .withStatement(String.format("insert into %s values(?, ?)", tableName))
+              .withPreparedStatementSetter(
+                  new JdbcIO.PreparedStatementSetter<KV<Integer, String>>() {
+                public void setParameters(
+                    KV<Integer, String> element, PreparedStatement statement) throws Exception {
+                  statement.setInt(1, element.getKey());
+                  statement.setString(2, element.getValue());
+                }
+              }));
+
+      pipeline.run();
+
+      try (Connection connection = dataSource.getConnection()) {
```

Comment:
Can you instead use JdbcIO.read() and PAssert for this? "Write-then-read" is the recommended style for tests of IOs.